### PR TITLE
fix(mercury): add client timestamp

### PR DESCRIPTION
**Pull request by chrisadubois**
# COMPLETES

## This pull request addresses
Clients have an issue with potentially ending up in an infinite reconnect loop. adding clientTimestamp to the socket allows mercury to throw away old mercury web sockets in the case of a thundering herd and limiting the potential for a 429 from mercury

## by making the following changes

Adding client timestamp to the socket open request

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios were tested

Automated unit tests

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request
- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

## Summary by CodeRabbit

- **New Features**
	- Enhanced connection management with improved error handling and logging.
	- Introduced a new `clientTimestamp` property for better connection attempt tracking.

- **Deprecations**
	- Deprecated `listen` and `stopListening` methods; users should now use `connect` and `disconnect`.

- **Bug Fixes**
	- Improved handling of connection errors and retries, ensuring robust connection behavior.

- **Tests**
	- Expanded test coverage for connection handling, error scenarios, and logout processes, including new tests for the `connect()` method and handling of provided URLs.

### CODE REVIEW

File: packages/@webex/internal-plugin-mercury/test/unit/spec/mercury.js [hunk 1]
```diff
@@ -448,7 +448,7 @@ describe('plugin-mercury', () => {
             assert.calledWith(
               Socket.prototype.open,
               sinon.match(/ws:\/\/providedurl.com/),
-              sinon.match.any
+              sinon.match.has('clientTimestamp', clock.now + 2) // 2 for 2 ticks
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-mercury/test/unit/spec/mercury.js]: where are the 2 ticks coming from? shouldn't we just mock the current time with `sinon.useFakeTimers()`?

Comment (chrisadubois) [reply packages/@webex/internal-plugin-mercury/test/unit/spec/mercury.js]: We can do that, but I'm using the clock that is defined above. The 2 ticks come from the beforeEach function above with `process.nextTick`. - was trying to utilize the code which already exists here.

Comment (marcin-bazyl) [reply packages/@webex/internal-plugin-mercury/test/unit/spec/mercury.js]: ah, ok, so yeah we are already using fake timers, it's a bit convoluted why +2, but I guess it's fine

**Review by marcin-bazyl — APPROVED** [review]

**Review by chburket — APPROVED** [review]
LGTM

File: packages/@webex/internal-plugin-mercury/test/unit/spec/mercury.js (hunk 1 above)
Comment (chrisadubois) [reply packages/@webex/internal-plugin-mercury/test/unit/spec/mercury.js]: I've changed this to just do one tick (for the mock websocket) and using fake timers

**Pull request merged**